core/src/main/resources/error/error-classes.json

```
@@ -586,6 +586,12 @@
       "Detected an incompatible DataSourceRegister. Please remove the incompatible library from classpath or upgrade it. Error: <message>"
     ]
   },
+  "INCOMPATIBLE_JOIN_TYPES" : {
+    "message" : [
+      "The join types <joinType1> and <joinType2> are incompatible."
+    ],
+    "sqlState" : "42818"
```

42818 is about operators (e.g. /, * +) and unctions.
```suggestion
    "sqlState" : "42613"
```

Note I think there is nothing wrong with a LATERAL on the RHS of of one of these joins.
SELECT * FROM T LEFT JOIN S RIGHT JOIN LATERAL(SELECT T.c1) .... should be legal